CRM/Mailing/Event/BAO/Opened.php

```diff
@@ -241,11 +241,24 @@ public static function &getRows(
     $contact = CRM_Contact_BAO_Contact::getTableName();
     $email = CRM_Core_BAO_Email::getTableName();
 
+    $selectClauses = array(
+      "$contact.display_name as display_name",
+      "$contact.id as contact_id",
+      "$email.email as email",
+      "{$open}.time_stamp as date",
+    );
+
+    if ($is_distinct) {
+      $groupBy = " GROUP BY $queue.id ";
+      $select = CRM_Contact_BAO_Query::appendAnyValueToSelect($selectClauses, "{$open}.time_stamp as date");
```

Comment:
I think too that, omitting other select fields will improve performance.

Earlier I was thinking of using max(time_stamp) to fetch the latest but I think here for distinct timestamp is a nondeterministic attribute? 

@lcdservices thoughts?